feed_test.go

```diff
@@ -84,3 +89,84 @@ func TestNotificationFeedBasic(t *testing.T) {
 		t.Fatal()
 	}
 }
+
+func TestUpdateActivities(t *testing.T) {
+	client, err := New(&Config{
+		APIKey:    os.Getenv("STREAM_API_KEY"),
+		APISecret: os.Getenv("STREAM_API_SECRET"),
+		AppID:     os.Getenv("STREAM_APP_ID"),
+		Location:  os.Getenv("STREAM_REGION"),
+	})
+	require.Nil(t, err)
+
+	feed, err := client.FlatFeed("flat", "123")
+	require.Nil(t, err)
+
+	tt, _ := time.Parse("2006-01-02T15:04:05.999999", "2017-08-31T08:29:21.151279")
+
+	testCases := []struct {
+		tt          time.Time
+		activities  []*Activity
+		shouldError bool
+	}{
+		{
+			activities: []*Activity{
+				&Activity{
+					ForeignID: "bob:123",
+					TimeStamp: &tt,
+					Actor:     "bob",
+					Verb:      "verb",
+					Object:    "object",
+				},
+			},
+			shouldError: false,
+		},
+		{
+			activities: []*Activity{
+				&Activity{
+					TimeStamp: &tt,
+					Actor:     "bob",
+					Verb:      "verb",
+					Object:    "object",
+				},
+			},
+			shouldError: true,
+		},
+		{
+			activities: []*Activity{
+				&Activity{
+					ForeignID: "bob:123",
+					Actor:     "bob",
+					Verb:      "verb",
+					Object:    "object",
+				},
+			},
+			shouldError: true,
+		},
+		{
+			activities:  []*Activity{},
+			shouldError: true,
+		},
+		{
+			activities: []*Activity{
+				&Activity{
+					ForeignID: "alice:123",
+					TimeStamp: &tt,
+					Actor:     "alice",
+					Verb:      "verb",
+					Object:    "object",
+				},
+			},
```

Comment:
I might be missing something, but this test seems to be a duplicate of the first one.

Reply:
as you can see, it has a different foreign id